events/tutorials/tutorial_import_user_events_gcs.md

```diff
@@ -0,0 +1,155 @@
+# **Import User Events From the Cloud Storage Tutorial**
+
+## Let's get started
+
+Retail API offers you an easy way to import your user event data from a Cloud Storage. All you need is to provide a name of
+the JSON file in the GCS bucket.
+
+This type of import is useful when you need to import a large amount of items to your catalog in a single step.
+
+You can find more information about different import types, their restrictions, and use cases in the [Retail API documentation](https://cloud.google.com/retail/docs/import-user-events)
+
+**Time to complete**: 
+<walkthrough-tutorial-duration duration="3.0"></walkthrough-tutorial-duration>
+
+## Before you begin
+
+To run Python code samples from this tutorial, you need to set up your virtual environment.
+
+To do that, run the following commands in a terminal:
+```bash
+pip install virtualenv
+```
+```bash
+virtualenv <your-env>
+```
+```bash
+source <your-env>/bin/activate
+```
+Next, install Google packages:
+```bash
+pip install google
+```
+```bash
+pip install google-cloud-retail
+```
+
+**Tip**: Click the copy button on the side of the code box to paste the command in the Cloud Shell terminal to
+run it.
+
+## Set the PROJECT_NUMBER environment variable
+
+As you are going to run the code samples in your own Cloud Project, you should specify the **project_id** as an environment variable, it will be used in every request to the Retail API.
+
+You can find the ```project_number``` in the **Home/Dashboard/Project Info card**.
+
+Set the environment variable with a following command:
+```bash
+export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+```
+
+## Import user events from the Cloud Storage source
+
+To upload catalog data to the Cloud Storage bucket, create one or more JSON files with user events that do not exceed 2 GB each. You can set up to 100 JSON files in a single import request. For more information, refer to the [example of the user event in JSON format](https://cloud.google.com/retail/docs/user-events#formats)
+
+To check the example of an import user events request, open **events/import_user_events_gcs.py**.
+
+The field **```parent```** contains a **catalog name** along with a branch number you are going to import your
+user events to.
+
+The field **```input_config```** defines the **GcsSource** as an import source.
```

Comment:
`The **```input_config```** field defines the **GcsSource** as an import source.`

and please check if the GCS should be uppercase.